packages/amplify_datastore_plugin_interface/lib/src/types/models/model_association.dart

```diff
@@ -20,22 +20,22 @@ enum ModelAssociationEnum { HasMany, HasOne, BelongsTo }
 
 class ModelAssociation {
   final ModelAssociationEnum associationType;
-  final String targetName; // opt (used in belongsTo)
-  final String associatedName; // opt (used in hasMany/hasOne)
-  final String associatedType; // opt (used in hasMany/hasOne)
+  final String? targetName; // opt (used in belongsTo)
+  final String? associatedName; // opt (used in hasMany/hasOne)
+  final String? associatedType; // opt (used in hasMany/hasOne)
 
   const ModelAssociation({
-    this.associationType,
+    required this.associationType,
     this.targetName,
     this.associatedName,
     this.associatedType,
   });
 
   ModelAssociation copyWith({
-    String name,
-    String targetName,
-    String associatedName,
-    String associatedType,
+    ModelAssociationEnum? associationType,

```

Comment:
Is this an existing bug being fixed here?

Reply:
What are you referring to here?  The types you be nullable: 

  ModelAssociation copyWith({
    ModelAssociationEnum? associationType,
    String? targetName,
    String? associatedName,
    String? associatedType,
  }) {